packages/@uppy/webcam/src/Webcam.tsx

```diff
@@ -1,73 +1,143 @@
 import { h } from 'preact'
 
 import { UIPlugin } from '@uppy/core'
+import type { Uppy, UIPluginOptions } from '@uppy/core'
+import type { DefinePluginOpts } from '@uppy/core/lib/BasePlugin.ts'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile.ts'
+import type { PluginTarget } from '@uppy/core/lib/UIPlugin.ts'
+import type { MinimalRequiredUppyFile } from '@uppy/core/lib/Uppy.ts'
 import getFileTypeExtension from '@uppy/utils/lib/getFileTypeExtension'
 import mimeTypes from '@uppy/utils/lib/mimeTypes'
 import isMobile from 'is-mobile'
 import canvasToBlob from '@uppy/utils/lib/canvasToBlob'
-import supportsMediaRecorder from './supportsMediaRecorder.js'
-import CameraIcon from './CameraIcon.jsx'
-import CameraScreen from './CameraScreen.jsx'
-import PermissionsScreen from './PermissionsScreen.jsx'
-
+import supportsMediaRecorder from './supportsMediaRecorder.ts'
+import CameraIcon from './CameraIcon.tsx'
+import CameraScreen from './CameraScreen.tsx'
+import PermissionsScreen from './PermissionsScreen.tsx'
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
-import locale from './locale.js'
+import locale from './locale.ts'
 
 /**
  * Normalize a MIME type or file extension into a MIME type.
  *
- * @param {string} fileType - MIME type or a file extension prefixed with `.`.
- * @returns {string|undefined} The MIME type or `undefined` if the fileType is an extension and is not known.
+ * @param fileType - MIME type or a file extension prefixed with `.`.
+ * @returns The MIME type or `undefined` if the fileType is an extension and is not known.
  */
-function toMimeType (fileType) {
+function toMimeType(fileType: string): string | undefined {
   if (fileType[0] === '.') {
-    return mimeTypes[fileType.slice(1)]
+    return mimeTypes[fileType.slice(1) as keyof typeof mimeTypes]
   }
   return fileType
 }
```

Comment:
I don't think this function does what the description says it does.  It could return any unknown string if the first argument doesn't start with `.` rather than `undefined`.

Instead of casting `fileTypes.slice(1)` to `keyof typeof mimeTypes` you could add a check to see if it is actually a key of `mimeTypes`

Reply:
> Instead of casting `fileTypes.slice(1)` to `keyof typeof mimeTypes` you could add a check to see if it is actually a key of `mimeTypes`

That sounds easier said than done 😅 In practice I don't think we'd get much better result from a strongly typed function (`fileType` is always a dynamic value), but I'd still be interested on how you could make it work if you have an idea.

Reply:
How about something like this?

```ts
function toMimeType(fileType: string): string | undefined {
  if (fileType[0] !== '.') return undefined;
  const key = fileType.slice(1)
  if (!(key in mimeTypes)) return undefined;
  return mimeTypes[key];
}
```

Reply:
extra points: 

```ts
 type MimeType = typeof mimeTypes[keyof typeof mimeTypes];

function toMimeType(fileType: string): MimeType | undefined {
  if (fileType[0] !== '.') return undefined;
  const key = fileType.slice(1)
  if (!(key in mimeTypes)) return undefined;
  return mimeTypes[key];
}
```

Reply:
Oh I thought you were suggesting changes in the types, rather than the implementation. No the current implementation is correct, it's meant to return whatever string was passed to it if it doesn't start with a dot.